crates/core_arch/src/x86/avx.rs

```diff
@@ -1425,7 +1425,7 @@ pub fn _mm256_insert_epi32<const INDEX: i32>(a: __m256i, i: i32) -> __m256i {
 /// [Intel's documentation](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_load_pd)
 #[inline]
 #[target_feature(enable = "avx")]
-#[cfg_attr(test, assert_instr(vmovap))]
+#[cfg_attr(all(test, not(target_env = "msvc")), assert_instr(vmovap))]
```

Comment:
Can you change these to be `all(target_arch = "x86", target_env = "msvc")` and add a comment pointing to https://github.com/rust-lang/rust/pull/139261 as the cause.